rewrite-test/src/main/kotlin/org/openrewrite/java/ChangeMethodAccessLevelTest.kt

```diff
@@ -0,0 +1,355 @@
+/*
+ * Copyright 2021 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java
+
+import org.junit.jupiter.api.Disabled
+import org.junit.jupiter.api.Test
+
+interface ChangeMethodAccessLevelTest : JavaRecipeTest {
+
+    @Test
+    fun fromPublicToPrivate(jp: JavaParser) = assertChanged(
+            jp,
+            recipe = ChangeMethodAccessLevel("com.abc.A aMethod(..)", "private"),
+            before = """
+            package com.abc;
+            class A {
+
+                @Deprecated // comment
+                public void aMethod(String s) {
+                }
+
+                // comment
+                @Deprecated
+                public void aMethod() {
+                }
+
+                // comment
+                public void aMethod(Integer i) {
+                }
+
+                public void aMethod(Double i) {
+                }
+            }
+        """,
+            after = """
+            package com.abc;
+            class A {
+
+                @Deprecated // comment
+                private void aMethod(String s) {
+                }
+
+                // comment
+                @Deprecated
+                private void aMethod() {
+                }
+
+                // comment
+                private void aMethod(Integer i) {
+                }
+
+                private void aMethod(Double i) {
+                }
+            }
+        """
+    )
+
+    @Test
+    fun fromPackagePrivateToProtected(jp: JavaParser) = assertChanged(
+            jp,
+            recipe = ChangeMethodAccessLevel("com.abc.A aMethod(..)", "protected"),
+            before = """
+            package com.abc;
+            class A {
+
+                void aMethod(Double i) {
+                }
+
+                @Deprecated // comment
+                void aMethod(String s) {
+                }
+            }
+        """,
+            after = """
+            package com.abc;
+            class A {
+
+                protected void aMethod(Double i) {
+                }
+
+                @Deprecated // comment
+                protected void aMethod(String s) {
+                }
+            }
+        """
+    )
+
+    @Disabled
+    @Test
+    fun fromPackagePrivateToProtected_broken(jp: JavaParser) = assertChanged(

```

Comment:
These are tests / situations that are currently not correctly handled.
Some (all?) are already not supported by code in existing visitors (see links in the ticket + changes in HideUtilityClassConstructorVisitor).
I spent quite some time on it, but I haven't succeed in fixing them (this is my first incursion with AST ^^)

I think this PR is already an improvement, in the sense that it'll factorize the logic, and that all other visitors using it will benefit from improvements and future fix to this visitor.